### ISSUE TITLE
Cannot find module '@mc-cdk/mc-efk'

### ISSUE THREAD

**Issue by posilva** [aws/aws-cdk#3790]
### The Question

I am integrating, for the first time, a custom construct in a stack. This construct is being installed with NPM from a git repository, added to the `package.json` dependencies: 
```
"dependencies": {
...
    "@mc-cdk/mc-efk": "git+ssh://git@stash.company.com/~user/mc-efk.git#master",
...

  }

```
when I run `cdk synth` the following error occurs:

```
Cannot find module '@mc-cdk/mc-efk'
Subprocess exited with error 1
```

**Any guidance of what I am missing or doing wrong?**

### Environment

  - **CDK CLI Version:**  1.5.0 (build c020efa)
  - **Module Version:** 
  - **OS:**  OSX High Sierra
  - **Language:**  TypeScript

### Other information

`npm run build` does not has any error.

**Comment by RomainMuller** [comment]
Might be our node dependency listing that blows up when modules are specified by `git` repository. This sounds odd though... It'll probably need further details (maybe consider attaching a verbose output?)

**Comment by posilva** [comment]
Removed some of the data used in the context:
this is the output of `cdk -v synth`

```
CDK toolkit version: 1.5.0 (build c020efa)
Command line arguments: { _: [ 'synth' ],
  'ignore-errors': false,
  ignoreErrors: false,
  json: false,
  j: false,
  verbose: true,
  v: true,
  ec2creds: undefined,
  i: undefined,
  'version-reporting': undefined,
  versionReporting: undefined,
  'path-metadata': true,
  pathMetadata: true,
  'asset-metadata': true,
  assetMetadata: true,
  staging: true,
  'role-arn': undefined,
  r: undefined,
  roleArn: undefined,
  '$0': '/usr/local/bin/cdk' }
Determining whether we're on an EC2 instance.
Does not look like EC2 instance.
cdk.json: {
  "context": {
    "key_pair": "keys1",
    "vpc_id": "vpc-00000",
    "ami": "ami-0cdab515472ca0bac",
    "acmCertARN": "arn:aws:acm:eu-central-1:000000000000:certificate/000000000000",
    "privateSubNets": [
      "subnet-000000000000",
      "subnet-000000000000",
      "subnet-000000000000"
    ],
    "publicSubNets": [
      "subnet-000000000000",
      "subnet-000000000000",
      "subnet-000000000000"
    ],
    "sshSG": "sg-000000000000",
    "EFKProduct": "some",
    "EFKEnvironment": "dev",
    "EFKVersion": "1.0",
    "EFKSharedSecret": "mysecret",
    "EFKPort": 5001
  },
  "app": "npx ts-node bin/somestack.ts"
}
merged settings: { versionReporting: true,
  pathMetadata: true,
  output: 'cdk.out',
  context:
   { key_pair: 'keys1',
     vpc_id: 'vpc-0000000',
     ami: 'ami-0cdab515472ca0bac',
     acmCertARN:
      'arn:aws:acm:eu-central-1:000000000000:certificate/000000000000',
     privateSubNets:
      [ 'subnet-000000000000',
        'subnet-000000000000',
        'subnet-000000000000' ],
     publicSubNets:
      [ 'subnet-000000000000',
        'subnet-000000000000',
        'subnet-000000000000' ],
     sshSG: 'sg-000000000000',
     EFKProduct: 'some',
     EFKEnvironment: 'dev',
     EFKVersion: '1.0',
     EFKSharedSecret: 'mysecret',
     EFKPort: 5001 },
  app: 'npx ts-node bin/somestack.ts',
  tags: [],
  assetMetadata: true,
  toolkitBucket: {},
  staging: true }
Setting "CDK_DEFAULT_REGION" environment variable to us-west-2
Resolving default credentials
Looking up default account ID from STS
Default account ID: 000000000000
Setting "CDK_DEFAULT_ACCOUNT" environment variable to 000000000000
context: { key_pair: 'keys1',
  vpc_id: 'vpc-df000000',
  ami: 'ami-0cdab515472ca0bac',
  acmCertARN:
   'arn:aws:acm:eu-central-1:0000000000:certificate/00000-00000-4a03-94b0-000000',
  privateSubNets:
   [ 'subnet-00000001',
     'subnet-00000002',
     'subnet-00000003' ],
  publicSubNets:
   [ 'subnet-00000004',
     'subnet-00000005',
     'subnet-00000006' ],
  sshSG: 'sg-0000000',
  EFKProduct: 'Some',
  EFKEnvironment: 'dev',
  EFKVersion: '1.0',
  EFKSharedSecret: 'mysecret',
  EFKPort: 5001,
  'aws:cdk:enable-path-metadata': true,
  'aws:cdk:enable-asset-metadata': true }
outdir: cdk.out
env: { CDK_DEFAULT_REGION: 'us-west-2',
  CDK_DEFAULT_ACCOUNT: '000000000000',
  CDK_CONTEXT_JSON:
   '{"key_pair":"keys1","vpc_id":"vpc-df000000","ami":"ami-0cdab515472ca0bac","acmCertARN":"arn:aws:acm:eu-central-1:00000:certificate/00000","privateSubNets":["subnet-0000","subnet-0000","subnet-99999"],"publicSubNets":["subnet-d2121212","subnet-ddsd132","subnet-0998989"],"sshSG":"sg-000000","EFKProduct":"some","EFKEnvironment":"dev","EFKVersion":"1.0","EFKSharedSecret":"mysecret","EFKPort":5001,"aws:cdk:enable-path-metadata":true,"aws:cdk:enable-asset-metadata":true}',
  CDK_OUTDIR: 'cdk.out' }
Cannot find module '@mc-cdk/mc-efk'
Subprocess exited with error 1
Error: Subprocess exited with error 1
    at ChildProcess.proc.on.code (/usr/local/lib/node_modules/aws-cdk/lib/api/cxapp/exec.ts:110:23)
    at ChildProcess.emit (events.js:182:13)
    at ChildProcess.EventEmitter.emit (domain.js:441:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
```

**Comment by shivlaks** [comment]
hey @posilva 

I'm trying to reproduce this scenario. Can you elaborate a little bit more on what `~user` expands into? is that just the GitHub username?

Can you confirm whether `npm install` works but `cdk synth` results in the issue mentioned here?

**Comment by posilva** [comment]
Hi Shiv, 

while I was trying npm install to answer your question, I faced other issues that lead me to  notice that the git repo does not have the .js (I had a .gitignore rule) files, after I have  rebuild it with jsii and committed the .js files I was able to solve the error.

I am not a TS developer (more backend developer), still "learning" the js world issues.

Thanks to take the time to look into it. Closing the ticket now
